www/class/centreonHost.class.php

```diff
@@ -405,40 +417,36 @@ public function getOneHostName($hostId)
     }
 
     /**
-     * @param array $hostId
-     * @return array
-     * @throws Exception
+     * @param int[] $hostId
+     * @return array $retArr
      */
-    public function getHostsNames($hostId = array())
-    {
-        $arrayReturn = array();
+    public function getHostsNames($hostId = []): array {
+        $retArr = [];
         $explodedValues = '';
         if (!empty($hostId)) {
-            $query = 'SELECT host_id, host_name ' .
-                'FROM host where host_id IN (';
-
-            for ($i = 1; $i <= count($hostId); $i++) {
-                $explodedValues .= '?,';
-            }
-            $explodedValues = rtrim($explodedValues, ',');
-            $hostId = array_map(
-                function ($var) {
-                    return (int)$var;
-                },
-                $hostId
-            );
-            $query .= $explodedValues . ') ';
-            $stmt = $this->db->prepare($query);
-            $dbResult = $stmt->execute($hostId);
-            if (!$dbResult) {
-                throw new \Exception("An error occured");
-            }
+            /* checking here that the array provided as parameter
+             * is exclusively made of integers (host ids)
+             */
+            $filteredHostIds = $this->filteredArrayId($hostId);
+            if (count($filteredHostIds) > 0) {
+                $query = 'SELECT host_id, host_name ' .
+                    'FROM host where host_id IN ( :hostIds )';
+                $stmt = $this->db->prepare($query);
+                $stmt->bindValue(':hostIds', implode(',', $filteredHostIds), \PDO::PARAM_STR);
```

Comment:
This will not work.
You need to bind each host id separately

Reply:
**Modifications done**